aptos-move/aptos-vm/src/sharded_block_executor/block_executor_client.rs

```diff
@@ -0,0 +1,62 @@
+// Copyright © Aptos Foundation
+
+use crate::block_executor::BlockAptosVM;
+use aptos_state_view::StateView;
+use aptos_types::transaction::{Transaction, TransactionOutput};
+use move_core_types::vm_status::VMStatus;
+use std::sync::Arc;
+
+pub trait BlockExecutorClient {
+    fn execute_block<S: StateView + Sync>(
+        &self,
+        transactions: Vec<Transaction>,
+        state_view: &S,
+        concurrency_level: usize,
+        maybe_block_gas_limit: Option<u64>,
+    ) -> Result<Vec<TransactionOutput>, VMStatus>;
+}
+
+impl BlockExecutorClient for LocalExecutorClient {
```

Comment:
Maybe only me, but I'm a bit confused by the naming: here the `Client` in `BlockExecutorClient` gives me the impression that anyone implementing it can talk to some service (in this case `ExecutorService`). Then `ExecutorService` holds a `LocalExecutorClient` which implements BlockExecutorClient...

IMO `BlockExecutor` is a better name.

Reply:
The issue is that `BlockExecutor` is already being used for the upper execution layer.  Naming is hard :(